src/ui/components/home/home_activity.rs

```diff
@@ -1,405 +1,52 @@
-        page_cursor: from,
-        show_hidden: None,
-      };
-
-      let result = LemmyClient.list_posts(form).await;
-
-      match result {
-        Ok(o) => Some(o),
-        Err(e) => {
-          error.set(Some(e));
-          None
-        }
-      }
-    },
-  );
-
-  #[cfg(not(feature = "ssr"))]
-  {
-    let iw = window()
-      .inner_width()
-      .ok()
-      .map(|b| b.as_f64().unwrap_or(0.0))
-      .unwrap_or(0.0);
-
-    let on_resize = move |_| {
-      let iw = window()
-        .inner_width()
-        .ok()
-        .map(|b| b.as_f64().unwrap_or(0.0))
-        .unwrap_or(0.0);
-
-      let mut query_params = query.get();
-      if iw >= 2560f64 {
-        query_params.insert("limit".into(), "30".to_string());
-      } else if iw >= 1536f64 {
-        query_params.insert("limit".into(), "20".to_string());
-      } else {
-        query_params.remove("limit");
-      }
-
-      if iw >= 640f64 {
-        csr_infinite_scroll_posts.set(None);
-        csr_paginator.set(None);
-      }
-
-      let navigate = leptos_router::use_navigate();
-      navigate(
-        &format!("{}", query_params.to_query_string()),
-        Default::default(),
-      );
-    };
-
-    window_event_listener_untyped("resize", on_resize);
-
-    if let Ok(e) = web_sys::Event::new("resize") {
-      on_resize(e);
-    }
-
-    if iw < 640f64 {
-      let on_scroll = move |_| {
-        let h = window()
-          .inner_height()
-          .ok()
-          .map(|b| b.as_f64().unwrap_or(0.0))
-          .unwrap_or(0.0);
-        let o = window().page_y_offset().ok().unwrap_or(0.0);
-        let b = f64::from(document().body().map(|b| b.offset_height()).unwrap_or(1));
-
-        let endOfPage = h + o >= b;
-
-        if endOfPage {
-          create_local_resource(
-            move || (user.get(), list_func(), sort_func()),
-            move |(_user, list_type, sort_type)| async move {
-              let form = GetPosts {
-                type_: list_type,
-                sort: sort_type,
-                community_name: None,
-                community_id: None,
-                page: None,
-                limit: None,
-                saved_only: None,
-                disliked_only: None,
-                liked_only: None,
-                page_cursor: csr_paginator.get(),
-                show_hidden: None,
-              };
-
-              let result = LemmyClient.list_posts(form).await;
-
-              match result {
-                Ok(mut o) => {
-                  csr_paginator.set(o.next_page);
-                  let mut p = csr_infinite_scroll_posts.get().unwrap_or(vec![]);
-                  p.append(&mut o.posts);
-                  csr_infinite_scroll_posts.set(Some(p));
-                }
-                Err(e) => {
-                  error.set(Some(e));
-                }
-              }
-            },
-          );
-        }
-      };
-
-      window_event_listener_untyped("scroll", on_scroll);
-    }
```

Comment:
I don't understand why we were using different fetch limits for different screen sizes. As for the infinite scroll logic, [there is a library with various leptos utility functions](https://docs.rs/leptos-use/latest/leptos_use/fn.use_infinite_scroll.html) that would greatly simplify things.

Reply:
more screen space, more content